### PR TITLE
Feature/859

### PR DESCRIPTION
**Pull request by emilnormann**
# Description

This Fixes #\<859> which is allowing the user to copy weekplans both for themselves and for other users. It also allows the user to copy activities in a specific weekplan.
The edit functionality for a weekplan is marginally related to the issue.
Much of the functionality was already implemented, this pull request makes it possible to copy multiple weekplans at once and fixes multiple bugs. 

Functionality: 

- You can edit a single weekplan at a time where you can change the name/week/year/pictogram of a weekplan.
- You can copy any amount of weekplans to any amount of users.
       -If a single weekplan is chosen you can copy to the already selected user which will make you edit the weekplan since it 
        will otherwise just be overwritten
        -If a single weekplan is chosen and you copy to other users you can edit the weekplan in case of any duplicates among 
         the chosen users
        -If multiple weekplans are copied at once you can only overwrite or cancel
- In a weekplan you can copy marked activities to chosen day(days)

Known bugs: 
There are two bugs related to the copying of activites both of them are in the web-api and therefore will not be fixed in this pull request

1.  When copying a choiceboard the name will not be kept (This is fixed in a pull request for the Web-api (https://github.com/aau-giraf/web-api/pull/278)
2. When copying an activity with an alarm the id of the alarm is copied and thus when one is deleted the other will be as well (Unfixed)

## Type of change
*Delete unchecked boxes (only for Type of change)*

- [ x] Bug fix (non-breaking change which fixes an issue)
- [x ] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

*Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration*

**Development Configuration**
*Type "flutter --version" and "dart --version" in your CMD to check versions.*

* Flutter version:
* Dart version:

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas, if necessary
- [ ] I have made corresponding changes to the documentation, if necessary
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works, if necessary
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have Acceptance Tested this on an iOS device
- [ ] I have Acceptance Tested this on an Android device

### CODE REVIEW

File: lib/screens/copy_to_citizens_screen.dart [hunk 1]
```diff
@@ -75,14 +77,23 @@ class CopyToCitizensScreen extends StatelessWidget {
...
+                                    _copiedWeekModelList.length > 1 ?
+                                    _showConflictDialogMultiplePlans(
+                                      context, conflicts,
+                                        _bloc.getAllConflictingUsers(
+                                            _currentUser, _copiedWeekModelList))
+                                        : _showConflictDialog(
+                                          context, conflicts,
+                                        _bloc.getAllConflictingUsers(
+                                            _currentUser,
+                                            _copiedWeekModelList));
```
Comment (ltranu20) [inline lib/screens/copy_to_citizens_screen.dart]: plz ingen conditional her, fuldstændig ulæsligt

File: lib/main.dart [hunk 2]
```diff
@@ -22,7 +22,7 @@ void main() {
    * Use the "environments.local.json" for running against your local web-api
    * For IOS users: change the SERVER_HOST in the environment.local file to "http://localhost:5000"
    */
-  environment.setFile('assets/environments.dev.json').whenComplete(() {
+  environment.setFile('assets/environments.local.json').whenComplete(() {
```
Comment (ltranu20) [inline lib/main.dart]: Meningen?

File: lib/screens/copy_to_citizens_screen.dart [hunk 3]
```diff
@@ -151,40 +162,93 @@ class CopyToCitizensScreen extends StatelessWidget {
...
+          barrierDismissible: false,
+          context: context,
+          builder: (BuildContext context) {
+            return Giraf3ButtonDialog(
+              title: 'Håndter konflikt',
+              description: userStringList.length < 500 ? '${userList.length == 1
+                  ? '' : 'Følgende borgere: '}'
+                  '$userStringList har i alt $conflicts '
+                  'konflikt${conflicts > 1 ? 'er' : ''}.' : '${userList.length}'
+                  'antal borgere har i alt $conflicts konflikt${conflicts > 1 ?
```
Comment (ltranu20) [inline lib/screens/copy_to_citizens_screen.dart:192]: Det her er også meget svær at finde hoved og hale i, men har ikke selv nogen ide hvordan det kan stå pænere ¯\\_(ツ)_/¯

**Review by ltranu20 — CHANGES_REQUESTED** [review]

File: lib/screens/copy_to_citizens_screen.dart (hunk 1 above)
Comment (emilnormann) [reply lib/screens/copy_to_citizens_screen.dart]: Går ud fra du mener ternary operator, har ændret det til if/else nu. Ved ikke om det er meget mere læsbart men ved ikke hvordan det ellers skulle skrives.

Comment (ltranu20) [reply lib/screens/copy_to_citizens_screen.dart]: yeah mb ja, men stadig lidt nemmere at se hvad tilhører hvad.

**Review by ltranu20 — APPROVED** [review]

**Review by manfredhod — APPROVED** [review]

**Pull request merged**